cli/lsp/language_server.rs

```diff
@@ -217,13 +283,23 @@ impl LanguageServer {
           check_js: false,
         },
       )?;
+
+      // Update the lockfile on the file system with anything new
+      // found after caching
+      if let Some(lockfile) = cli_options.maybe_lockfile() {
+        let lockfile = lockfile.lock();
+        if let Err(err) = lockfile.write() {
+          lsp_warn!("Error writing lockfile: {}", err);
+        }
+      }
```

Comment:
This is great to have, but general comment is that we should figure out if we could remove stale entries from the lockfile. Otherwise it feels that quickly iterating on a project in the LSP will just keep growing the lockfile indefinitely.

Reply:
Since we can have multiple entrypoints, the only way to do it that I can think of would be to store which local files have what external modules, then once there are no more references it can be removed from the lockfile. This would be kind of annoying because anytime you'd add a reference in a file, then it would need to update the lockfile as well. Additionally, it would need to handle when files are deleted. Overall, I think that's more annoying than an indefinitely growing lockfile. People can have a task that resets it or they can just delete it periodically when upgrading their dependencies.

Reply:
Fair enough, IIRC there's already an issue requesting such feature.